src/mapper/pkg/otelexporter/otel_exporter.go

```diff
@@ -0,0 +1,140 @@
+package otelexporter
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/otterize/network-mapper/src/mapper/pkg/intentsstore"
+	sharedconfig "github.com/otterize/network-mapper/src/shared/config"
+	"github.com/otterize/network-mapper/src/shared/version"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+)
+
+type OtelExporter struct {
+	config        Config
+	counter       metric.Int64Counter
+	intentsHolder *intentsstore.IntentsHolder
+}
+
+func newResource() (*resource.Resource, error) {
+	return resource.Merge(resource.Default(),
+		resource.NewWithAttributes(semconv.SchemaURL,
+			semconv.OTelLibraryName("otterize/network-mapper"),
+			semconv.OTelLibraryVersion(version.Version()),
+		))
+}
+
+const DefaultMetricEndpoint = "ingest.lightstep.com:443"
+
+// uses same name as expected in opentelemetry-collector-contrib's servicegraphprocessor
+const CounterMetricName = "traces_service_graph_request_total"
+const ClientAttributeName = "client"
+const ServerAttributeName = "server"
+
+func getenv(key, fallback string) string {
+	value := os.Getenv(key)
+	if len(value) == 0 {
+		return fallback
+	}
+	return value
+}
+
+func newMeterProvider(ctx context.Context, res *resource.Resource, exportInterval time.Duration) (*sdk.MeterProvider, error) {
+	metricExporter, err := otlpmetricgrpc.New(ctx,
+		otlpmetricgrpc.WithEndpoint(getenv("OTEL_EXPORTER_OTLP_ENDPOINT", DefaultMetricEndpoint)),
+		otlpmetricgrpc.WithHeaders(map[string]string{
+			"lightstep-access-token": os.Getenv("LS_ACCESS_TOKEN"),
```

Comment:
Since this isn't OpenTelemetry specific, could we turn this into a generic configuration for additional OTel headers? Use `viper StringSlice` for this, and then the Otterize network mapper Helm chart could accept a list of additional headers.

In Lightstep's install instructions, you would pass an additional Helm chart value, like so: `--set networkMapper.otelHeaders.lightstep-access-token=<LS_ACCESS_TOKEN>`. Since you are passing your access token this way, we should also add the ability to get the value of an otelHeader specified this way from a Kubernetes Secret.

Reply:
Updated, there is also a special environment variable for header values as well.